### PR TITLE
Fixed issue with multiple entries in the taggable table

### PR DESCRIPTION
**Pull request by AlexVanderbist**
- Added a test to check for duplicate tag entries
- Check if pivot table entry exists before attaching tag to model

### CODE REVIEW

File: src/HasTags.php [hunk 1]
```diff
@@ -101,7 +101,9 @@ public function attachTags($tags)
         $tags = $className::findOrCreate($tags);
 
         collect($tags)->each(function (Tag $tag) {
-            $this->tags()->attach($tag);
+            if (! $this->tags()->get()->contains('id', $tag->id)) {
```
Comment (sebastiandedeyne) [inline src/HasTags.php]: Could we use `sync` here?

Comment (AlexVanderbist) [inline src/HasTags.php]: I've updated the PR to use `syncWithoutDetaching`. Thanks!

File: src/HasTags.php [hunk 2]
```diff
@@ -100,11 +100,11 @@ public function attachTags($tags)
 
         $tags = $className::findOrCreate($tags);
 
-        collect($tags)->each(function (Tag $tag) {
-            if (! $this->tags()->get()->contains('id', $tag->id)) {
-                $this->tags()->attach($tag);
-            }
-        });
+        if (! $tags instanceof \Illuminate\Support\Collection) {
```
Comment (sebastiandedeyne) [inline src/HasTags.php]: Doesn't the `Collection` class already do this check internally?

Comment (AlexVanderbist) [inline src/HasTags.php]: It does, I've removed the if-statement

**Comment by sebastiandedeyne** [comment]
Cool, thanks!

**Pull request merged**